### PR TITLE
Auto-kol smart contract integration

### PR DESCRIPTION
**Pull request by Xm0onh**
This PR address issue #38 and #27

### CODE REVIEW

File: auto-kol/agent/src/utils/sc.ts
Comment (jfrank-summit) [inline auto-kol/agent/src/utils/sc.ts:1]: we should give this a more descriptive name than `sc.ts`

File: auto-kol/agent/src/utils/sc.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    return await contract.getLastMemoryHash(wallet.address);
+}
+
+export async function setLastMemoryHash(hash: string) {
+    const tx = await contract.setLastMemoryHash(hash);
+    const receipt = await tx.wait();
+    return receipt;
+}
+
+export function watchMemoryHashUpdates(
```
Comment (jfrank-summit) [inline auto-kol/agent/src/utils/sc.ts:21]: It doesn't appear this is used anywhere, were you planning on adding it somewhere?

Comment (Xm0onh) [reply auto-kol/agent/src/utils/sc.ts:21]: Not yet, I kept it as placeholder when we want to add tools for resurrection.

File: auto-kol/agent/src/utils/dsn.ts [hunk 2]
```diff
@@ -1,20 +1,30 @@
 import { createLogger } from '../utils/logger.js';
 import { createAutoDriveApi, uploadFile } from '@autonomys/auto-drive';
-import { stringToCid, blake3HashFromCid } from '@autonomys/auto-dag-data';
+import { stringToCid, blake3HashFromCid, cidFromBlakeHash } from '@autonomys/auto-dag-data';
 import { addDsn } from '../database/index.js';
 import { v4 as generateId } from 'uuid';
 import { config } from '../config/index.js';
+import { setLastMemoryHash } from './sc.js';
+import { toHex, fromHex } from 'viem'
```
[the remark this answers is not in the document]
Comment (Xm0onh) [reply auto-kol/agent/src/utils/dsn.ts]: `fromHex` and `toHex` of viem library are written much better that etherjs. However, I removed `viem` completly for now. We can use them we integrated our chain info into their lib

**Review by jfrank-summit — APPROVED** [review]
LGTM

**Pull request merged**